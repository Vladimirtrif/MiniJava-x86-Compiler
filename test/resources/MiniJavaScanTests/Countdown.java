public class Countdown {
    public static void main(String[] args) {
        printCountdown(5);
    }

    public void printCountdown(int n) {
        int count = 0;
        while(count < n) {
            System.out.println(count);
            count = count + 1;
        }
        //Comment should be ignored, display should be identifier
        int display = 5;
        /*
        *** this comment should also be ignored
        // print should be identifier token as well
         */

        int print = -1;
    }
}