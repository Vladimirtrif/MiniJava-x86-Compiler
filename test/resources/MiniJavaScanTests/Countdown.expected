PUBLIC CLASS ID(Countdown) LBRACE PUBLIC STATIC VOID MAIN LPAREN STRING LBRACKET RBRACKET ID(args) RPAREN LBRACE ID(printCountdown) LPAREN INTEGER(5) RPAREN SEMICOLON RBRACE PUBLIC VOID ID(printCountdown) LPAREN INT ID(n) RPAREN LBRACE INT ID(count) BECOMES INTEGER(0) SEMICOLON WHILE LPAREN ID(count) LESS ID(n) RPAREN LBRACE PRINT LPAREN ID(count) RPAREN SEMICOLON ID(count) BECOMES ID(count) PLUS INTEGER(1) SEMICOLON RBRACE INT ID(display) BECOMES INTEGER(5) SEMICOLON INT ID(print) BECOMES MINUS INTEGER(1) SEMICOLON RBRACE RBRACE